keras_cv/utils/preset_utils.py

```diff
@@ -217,5 +220,20 @@ def legacy_load_weights(layer, weights_path):
             functional_cls = cls
     property = functional_cls._layer_checkpoint_dependencies
     functional_cls._layer_checkpoint_dependencies = {}
+
+    from keras_cv.models.task import Task
+
+    if not backend_config.keras_3() and isinstance(layer, Task):
+        # Hacky fix for Keras 2 backwards compatibility. Keras 2 traverses
+        # loading weights in the reverse order, causing a naming mismatch when
+        # loading Kaggle weights saved from Keras 3.
+        f = h5py.File(weights_path, "r+")
```

Comment:
Is it safe to modify the local file on disk (since it will affect subsequent usages, maybe using Keras 3 this time)? Could the renaming be done in memory?

Reply:
Renaming in memory will require changes to the saving APIs in Keras 2 itself, which we can't modify. However, this modifies the local file to be in the correct format (and is a one time fix).
On subsequent usages, no fix will be necessary here since "_backbone" will not exist.

Reply:
But then you can't reload the file in Keras 3, right? Would it be better to rename the local file then, so that it only gets used with Keras 2?

Reply:
Good idea, just added renaming logic for the local file thanks! Agreed, it is indeed safer this way.